### DISCUSSION TITLE
LTSP_CLIENT cookies IP address

### DISCUSSION THREAD

**Discussion by Valiumlord** [ltsp/ltsp#737] [Q&A] [answered]
Hello,
Its been awhile but i got everything running within the LTSP: booting works, clients log in automatically, google chrome starts up automatically.
But i've gotten into a problem. Since our intranet uses cookies to make statistic on work registration per pc, the new ltsp server that i've set up only shows server address if i inspect our intranet, not clients IP. Is there any workarounds or ? 

The terminals, that are in use at the moment, are run on LTSP5 on phisical server. I was given task to install LTSP server on Proxmox so we could shut down the old LTSP5. The cookie thing is the last problem.

**Comment by alkisg** [comment]
Hi, if I understood you correctly, your new server is running LTSP5.
LTSP5 isn't supported here, please see https://github.com/ltsp/ltsp5/wiki instead.

The commands and methodology for the new LTSP are different than LTSP5. Mixing both of them in the same discussions site would confuse users.

**Reply by Valiumlord** [reply]
Hello, 
Current server is running on LTSP5 and the new one im doing is on LTSP 22.01.

i'll try to explain my problem again (english is not my first language)
When the client boots and logs in, google chrome will automatically start and direct to certain intranet page. In our php code, we have a line that reads LTSP_CLIENT cookie ip address. Lets say my server NIC1 aadress is x.x.x.x and NIC2 address 192.168.67.xxx and client aadress is 192.168.67.x.
As i view page source on clients browser it shows "LTSP_CLIENT COOKIE: x.x.x.x"(which is server address) and i need it to show me "LTSP_CLIENT COOKIE: 192.168.67.x" (which is CLIENT address)

**Comment by alkisg** [comment]
In that LTSP client, press Ctrl+Alt+T to open a terminal.
Execute the following commands, and then paste the output here:

```shell
cat /proc/cmdline
ip a
```

**Reply by Valiumlord** [reply]
Sorry for late answer, been busy
cat /proc/cmdline
`initrd=ltsp.img initrd=initrd.img root=/dev/nfs nfsroot=192.168.67.200:/srv/ltsp ltsp.image=images/x86_64.img loop.max_part=9 BOOTIF=01-e0-d5-5e-c1-7c-6c
`
ip a
```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever

2: enp2s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether e0:d5:5e:c1:7c:6c brd ff:ff:ff:ff:ff:ff
    inet 192.168.67.201/24 brd 192.168.67.255 scope global enp2s0
       valid_lft forever preferred_lft forever
    inet6 fe80::e2d5:5eff:fec1:7c6c/64 scope link 
       valid_lft forever preferred_lft forever

3: wlp1s0: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN group default qlen 1000
    link/ether 0c:54:15:74:e0:fa brd ff:ff:ff:ff:ff:ff
```

**Comment by alkisg** [comment, marked as answer]
I think you mean that you're using the dual-NIC LTSP setup and that your clients are behind NAT. So, when they're accessing the non-LTSP local network, they're doing so via the LTSP server IP.
That's perfectly normal, it's how NAT works.

To solve the "problem", you may:

1. Put your web server inside the LTSP network (192.168.67.x) instead of outside it.
2. Or avoid using an LTSP network, and merge everything into one subnet, e.g. 10.0.0.x.
3. Or change your start page so that it includes the IP, e.g. http://my.web.server/my-start-page/index.php?ip=$IP&mac=$MAC.
